strategy/from_json/to_object_containing.ts

```diff
@@ -0,0 +1,55 @@
+// Copyright 2018-2020 Gamebridge.ai authors. All rights reserved. MIT license.
+
+import { JSONArray, JSONValue, Serializable } from "../../serializable.ts";
+import { FromJSONStrategy } from "../compose_strategy.ts";
+import {
+  ERROR_TO_OBJECT_CONTAINING_INVALID_SUB_VALUE,
+  ERROR_TO_OBJECT_CONTAINING_INVALID_VALUE,
+} from "../../error_messages.ts";
+import { getNew, isObject, StrategyTypeArgument } from "../_utils.ts";
+
+/** revive data from `{k: v}` using `fromJSON` on a subclass type `v` */
+export function toObjectContaining<T>(
+  type: StrategyTypeArgument<T>,
+): FromJSONStrategy {
+  return function _toObjectContaining(
+    value: JSONValue,
+  ) {
+    if (value == null) {
+      return null;
+    }
+
+    if (!isObject(value)) {
+      throw new Error(ERROR_TO_OBJECT_CONTAINING_INVALID_VALUE);
+    }
+
+    const record: Record<string, Serializable | Serializable[] | null> = {};
+    // check that JSONValue is something we can deal with
+    // but mostly to make the type checker happy
+    if (typeof value === "object" && !Array.isArray(value)) {
+      for (const prop in value) {
+        // null is a JSONValue
+        if (value[prop] === null) {
+          record[prop] = null;
+          continue;
+        }
+        if (Array.isArray(value[prop])) {
+          record[prop] = (value[prop] as JSONArray).map((v: JSONValue) => {
+            if (!isObject(v)) {
+              throw new Error(ERROR_TO_OBJECT_CONTAINING_INVALID_SUB_VALUE);
+            }
+            return getNew(type).fromJSON(v);
+          });
+          continue;
```

Comment:
How would you feel about changing the `if / continue / if / continue` to `if / else if ... / else if ... / else`? I watched a talk (from a C++ conference of all things) where someone was explaining why that can be easier to read (makes the control structure obvious just by looking at the indentation, basically).

Reply:
I am a fan of exit early exit often. We're not doing crazy processing so the shortcircuit seems appropriate.